### PR TITLE
[Proposed]More ore dictionary compatibility.

### PR DESCRIPTION
**Pull request by knoxx093**
Moved the normal oredict enabled addRecipe into the smelter and campfire class since other classes have it. Renamed an awkward function.

Went through the recipes and added oredict support wherever possible.
I'm not sure if any of these recipes were purposely incompatible.

Proposed changes, reject if unwanted.

### CODE REVIEW

File: src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java [hunk 1]
```diff
@@ -535,7 +538,7 @@ public static void addAdvancedRecipes()
 				" i ", "iQi", "###",
 				'i', Blocks.redstone_torch,
 				'Q', Items.quartz,
-				'#', Blocks.stone);
+				'#', "stone");
```
Comment (einsteinsci) [inline src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java]: Is `"stone"` the correct ore dictionary name for `Blocks.stone`?

File: src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java [hunk 2]
```diff
@@ -1247,32 +1250,32 @@ public static void addShapedRecipes()
...
+				'#', "cobblestone", '-', new ItemStack(Blocks.stone_slab, 1, 3))); // cobblestone slab
 		}
 
 		// Gravel->Flint
 		GameRegistry.addShapedRecipe(new ItemStack(Items.flint), "##", "##", '#', Blocks.gravel);
 
 		// Iron Nugget->Ingot
-		GameRegistry.addShapedRecipe(new ItemStack(Items.iron_ingot), "***", "***", "***", '*',
-									 RegisterItems.ironNugget);
+		GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(Items.iron_ingot), "***", "***", "***", '*',
```
Comment (einsteinsci) [inline src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java]: This is purposely specific to BB iron nugget. We don't want to be adding in recipes that other mods should be adding in themselves.

**Comment by einsteinsci** [comment]
You might want to re-submit this later, I have to pull the big restructuring fork from Leviathan143. The only problem I see is with the iron nugget to ingot recipe.

File: src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java (hunk 1 above)
Comment (knoxx093) [inline src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java]: It is.

File: src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java (hunk 2 above)
Comment (knoxx093) [inline src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java]: I'd thought that might be the case there.

What do you think about the reverse "ingotIron" to RegisterItems.ironNugget?

**Comment by knoxx093** [comment]
I believe this doesn't alter any of the same files as Leviathans restructuring, so you should just be able to merge it in when convenient. If this isn't the case I can rebase it later.

File: src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java [hunk 3]
```diff
@@ -35,7 +40,7 @@ public static void addShapelessRecipes()
 		GameRegistry.addShapelessRecipe(new ItemStack(Items.dye, 2, 15), RegisterItems.boneShard);
 
 		// Iron Nugget
-		GameRegistry.addShapelessRecipe(new ItemStack(RegisterItems.ironNugget, 9), Items.iron_ingot);
+		GameRegistry.addRecipe(new ShapelessOreRecipe(new ItemStack(RegisterItems.ironNugget, 9), "ingotIron"));
```
Comment (knoxx093) [inline src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java]: Is this good?

File: src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java (hunk 2 above)
Comment (einsteinsci) [inline src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java]: The reverse is fine. If a mod adds iron ingots, they might not necessarily add iron nuggets. There was a mod in FTB Monster that did this, can't remember what it was called. It added some kind of nether iron ingot registered under "ingotIron".

File: src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java (hunk 3 above)
Comment (einsteinsci) [inline src/main/java/net/einsteinsci/betterbeginnings/register/RegisterRecipes.java]: Yes, this is fine.

**Pull request merged**